src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/impl/ObjectionService.java

```diff
@@ -196,10 +198,41 @@ public void deleteAttachment(String requestId, String companyNumber, String obje
                 () -> new AttachmentNotFoundException(String.format(ATTACHMENT_NOT_FOUND_MESSAGE, attachmentId))
         );
 
-        attachments.remove(attachment);
+        boolean successfulDelete = deleteFromS3(requestId, companyNumber, objectionId, attachmentId);
 
-        objection.setAttachments(attachments);
+        if(successfulDelete) {
+            attachments.remove(attachment);
 
-        objectionRepository.save(objection);
+            objection.setAttachments(attachments);
+
+            objectionRepository.save(objection);
+        }
+    }
+
+    private boolean deleteFromS3(String requestId, String companyNumber, String objectionId, String attachmentId) {
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(LogConstants.COMPANY_NUMBER.getValue(), companyNumber);
+        logMap.put(LogConstants.OBJECTION_ID.getValue(), objectionId);
+        final String errorMessage = "Unable to delete attachment %s, status code %s";
```

Comment:
Would suggest defining these formatted error messages as class constants rather than creating new strings each time the method is called. A bit like in the FileTransferApiClient class above.